source/common/http/conn_manager_impl.cc

```diff
@@ -1176,8 +1176,14 @@ void ConnectionManagerImpl::ActiveStream::refreshDurationTimeout() {
   auto grpc_timeout = Grpc::Common::getGrpcTimeout(*request_headers_);
   std::chrono::milliseconds timeout;
   bool disable_timer = false;
-
-  if (!grpc_timeout || !route->grpcTimeoutHeaderMax()) {
+  if (route->usingNewTimeouts()) {
+    // If we are using new timeouts and MaxStreamDuration is not set, use the route's timeout
+    // value. This is the case when route's MaxStreamDuration is specified without inner
+    // MaxStreamDuration i.e. only grpc_timeout_header_max is set.
+    if (!route->maxStreamDuration()) {
```

Comment:
so if grpc_timeout_header_max is set when do we use it?  usingNewTimeouts would be true and we'd ignore all the code below, no?

Can you please fix and regression test?

Reply:
Oh, Yeah.. Missed that. Will add a test

Reply:
@alyssawilk Can you PTAL at the fix to see if this is correct? One thing I found was this needs changes to most of the test cases because now we will be initiating request timer most of the times. So I just want to ensure this is the right direction before I make changes the tests(some of which I have already done - please take a look at the changes)